pkg/vmcp/health/monitor_test.go

```diff
@@ -0,0 +1,533 @@
+package health
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"github.com/stacklok/toolhive/pkg/vmcp"
+	"github.com/stacklok/toolhive/pkg/vmcp/mocks"
+)
+
+func TestNewMonitor_Validation(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+	t.Cleanup(ctrl.Finish)
+
+	mockClient := mocks.NewMockBackendClient(ctrl)
+	backends := []vmcp.Backend{
+		{ID: "backend-1", Name: "Backend 1", BaseURL: "http://localhost:8080"},
+	}
+
+	tests := []struct {
+		name        string
+		config      MonitorConfig
+		expectError bool
+	}{
+		{
+			name: "valid config",
+			config: MonitorConfig{
+				CheckInterval:      30 * time.Second,
+				UnhealthyThreshold: 3,
+				Timeout:            10 * time.Second,
+			},
+			expectError: false,
+		},
+		{
+			name: "invalid check interval",
+			config: MonitorConfig{
+				CheckInterval:      0,
+				UnhealthyThreshold: 3,
+				Timeout:            10 * time.Second,
+			},
+			expectError: true,
+		},
+		{
+			name: "invalid unhealthy threshold",
+			config: MonitorConfig{
+				CheckInterval:      30 * time.Second,
+				UnhealthyThreshold: 0,
+				Timeout:            10 * time.Second,
+			},
+			expectError: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			monitor, err := NewMonitor(mockClient, backends, tt.config)
+			if tt.expectError {
+				assert.Error(t, err)
+				assert.Nil(t, monitor)
+			} else {
+				assert.NoError(t, err)
+				assert.NotNil(t, monitor)
+			}
+		})
+	}
+}
+
+func TestMonitor_StartStop(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockClient := mocks.NewMockBackendClient(ctrl)
+	backends := []vmcp.Backend{
+		{ID: "backend-1", Name: "Backend 1", BaseURL: "http://localhost:8080", TransportType: "sse"},
+	}
+
+	config := MonitorConfig{
+		CheckInterval:      100 * time.Millisecond,
+		UnhealthyThreshold: 3,
+		Timeout:            50 * time.Millisecond,
+	}
+
+	// Mock health check calls
+	mockClient.EXPECT().
+		ListCapabilities(gomock.Any(), gomock.Any()).
+		Return(&vmcp.CapabilityList{}, nil).
+		AnyTimes()
+
+	monitor, err := NewMonitor(mockClient, backends, config)
+	require.NoError(t, err)
+
+	// Start monitor
+	ctx := context.Background()
+	err = monitor.Start(ctx)
+	require.NoError(t, err)
+
+	// Wait for at least one health check
+	time.Sleep(150 * time.Millisecond)
```

Comment:
These `time.Sleep()` calls with tight margins might get flaky on slow CI. Worth keeping an eye on - if we see intermittent failures we could switch to polling or a mock clock.